bgpd/bgp_label.c

```diff
@@ -319,13 +329,13 @@ bgp_nlri_parse_label (struct peer *peer, struct attr *attr,
 
       if (attr)
         {
-          bgp_update (peer, &p, addpath_id, attr, packet->afi, SAFI_LABELED_UNICAST,
-                      ZEBRA_ROUTE_BGP, BGP_ROUTE_NORMAL, NULL, label, 0, NULL);
+          bgp_update (peer, &p, addpath_id, attr, packet->afi, SAFI_UNICAST,
+                      ZEBRA_ROUTE_BGP, BGP_ROUTE_NORMAL, NULL, &label, 0, NULL);
         }
       else
         {
-          bgp_withdraw (peer, &p, addpath_id, attr, packet->afi, SAFI_LABELED_UNICAST,
-                        ZEBRA_ROUTE_BGP, BGP_ROUTE_NORMAL, NULL, label, NULL);
+          bgp_withdraw (peer, &p, addpath_id, attr, packet->afi, SAFI_UNICAST,
```

Comment:
I must be missing something: why call update on a different SAFI than the one being processed?